### PR TITLE
Blobstore API with Google Cloud Storage samples

### PR DESCRIPTION
**Pull request by ryanmats**
Moved Blobstore API samples to api folder.
Added "Blobstore API with Google Cloud Storage" samples (taken from https://cloud.google.com/appengine/docs/python/blobstore/ and modified to work) to gcs folder

### CODE REVIEW

File: appengine/standard/blobstore/gcs/main.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+import cloudstorage
+from google.appengine.api import app_identity
+from google.appengine.ext import blobstore
+from google.appengine.ext.webapp import blobstore_handlers
+import webapp2
+
+
+# This handler creates a file in Cloud Storage using the cloudstorage
+# client library and then reads the data back using the Blobstore API.
+
```
Comment (jonparrott) [inline appengine/standard/blobstore/gcs/main.py]: Unnecessary blank line.

File: appengine/standard/blobstore/gcs/main.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+class CreateAndReadFileHandler(webapp2.RequestHandler):
+    def get(self):
+        # Get the default Cloud Storage Bucket name and create a file name for
+        # the object in Cloud Storage.
+        bucket = app_identity.get_default_gcs_bucket_name()
+
+        # Cloud Storage file names are in the format /bucket/object.
+        filename = '/{}/blobstore_demo'.format(bucket)
+
+        # Create a file in Google Cloud Storage and write something to it
```
Comment (jonparrott) [inline appengine/standard/blobstore/gcs/main.py]: nit: sentence needs a period. (throughout please)

File: appengine/standard/blobstore/gcs/main.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        self.response.headers['Content-Type'] = 'text/plain'
+        self.response.write(data)
+
+        # Delete the file from Google Cloud Storage using the blob_key.
+        blobstore.delete(blob_key)
+
+
+# This handler creates a file in Cloud Storage using the cloudstorage
+# client library and then serves the file back using the Blobstore API.
+
```
Comment (jonparrott) [inline appengine/standard/blobstore/gcs/main.py]: Unnecessary blank line.

File: appengine/standard/blobstore/gcs/main.py (hunk 2 above)
Comment (ryanmats) [inline appengine/standard/blobstore/gcs/main.py]: done

File: appengine/standard/blobstore/gcs/main.py (hunk 3 above)
Comment (ryanmats) [inline appengine/standard/blobstore/gcs/main.py]: done

**Pull request merged**